### ISSUE TITLE
Helix keymap

### ISSUE THREAD

**Issue by humb1t** [zed-industries/zed#4642]
I know it can be tricky, but the only reason preventing me from day to day use of zed is Helix with their beautiful system based on keys without mouse, multiple selections and other features.

Will be great to support that and maybe I will try to research possible ways to do so:
https://docs.helix-editor.com/keymap.html

**Comment by nyabinary** [comment]
Zed + Helix would be extremely nice :3

**Comment by alphabitserial** [comment]
I would absolutely love to have this. I've been using Helix as my main editor for several months. Having a GUI with the same ergonomics would be fantastic.

**Comment by seanchen1991** [comment]
Now that the repository has been open-sourced, what would be a rough outline of adding Helix keymap support? 🙂

**Comment by danriedl** [comment]
> I would absolutely love to have this. I've been using Helix as my main editor for several months. Having a GUI with the same ergonomics would be fantastic.

I'm in the same spot but have to say, that i'll stick to helix itself.
Tried helix - binding in vscode but still it's not the same experience.

**Comment by ascarter** [comment]
@JosephTLyons I think the request here would be to do a modal keymap that is `Selection-Action` instead of `Action-Selection`. There are multiple modal editors now (which Helix is probably one of the best) that use this model. If Zed could support it, would be really cool to see.

Another thought long term - NeoVim crew embedded their engine into VS Code. You can run the editor window using NeoVim but within the rest of VS Code. Would be very interesting for Zed to do something like this where another terminal type engine could power the editor. Maybe a chance to get the best of both worlds.

**Comment by andrew-flick** [comment]
I would like to look into this as I use Helix in preference to vim when I can. But how to do it? Is it a complete new mode? Or is it a key-binding hack on vim-mode?

**Comment by andreweades** [comment]
I would like to look into this as I use Helix in preference to vim when I can. But how to do it? Is it a complete new mode? Or is it a key-binding hack on vim-mode?

**Comment by clo4** [comment]
@andreweades You'd have to introduce a new `helix` crate to implement some of the actions, and create a new keymap to use the actions. It's distinct enough from Vim that it would probably warrant a new editing mode, because selection->action really is a distinct paradigm from action->motion. But it might be worth taking a step back and instead of just copying the Helix keymap thinking about what it might look like to design a selection->action keymap *just for Zed*. Most of the infrastructure is already there! Would it feel different from Kakoune and Helix given it's in a GUI editor?

**Comment by wmstack** [comment]
Also I do like the helix bindings but I don't like to retain the selection when returning using insert mode or append mode. I have this mapped in helix `a = ['append_mode', 'collapse_selection']` and `i = ['insert_mode', 'collapse_selection']`. I hope similar bindings can be supported in helix.

Also vim doesn't really allow you to select the newline character. Helix does. I have a hard time differentiating a whitespace from a newline in helix so I hope the cursor changes colour or something when it is on a newline character.

**Comment by WarrenAdams8** [comment]
+1

**Comment by andreweades** [comment]
@clo4 Good point. I use a custom Helix keymap as some of my muscle memory from VIM does not translate well. I try to limit the amount of mousing as after 30+ years of mousing I have RSI in my shoulder so I like editors that have VIM and now Helix keys. I think selection->action makes a lot of sense so I would love to see it in Zed at some point and would not mind contributing when I have time.

**Comment by wmstack** [comment]
@JosephTLyons I wonder why this is not in the top ranking issues? It has more votes than a lot of the ones I see there. Is it because it doesn't neatly fit into one of the categories? Maybe it should be in the vim section.

**Comment by JosephTLyons** [comment]
> @JosephTLyons I wonder why this is not in the top ranking issues? It has more votes than a lot of the ones I see there. Is it because it doesn't neatly fit into one of the categories? Maybe it should be in the vim section.

This would only be included on the top-ranking issues report if it had more likes than the item at the bottom of the `enhancement` section (`Git in-line blame indicators` - 81 👍).

**Comment by zoedsoupe** [comment]
i would love to have helix support inside zed 💜

how i could contribute to achieve this new crate? i'm not the very best rust developer but i think i can mimic the vim crate somehow and do some tweaks to achieve `selection->action` mode.

**Comment by ConradIrwin** [comment]
@zoedsoupe There's some work at #11130, but the original author is unable to keep working on it. If you'd like to start working on that branch that would be very appreciated!

We decided to stay in the same crate as vim for now (because that way we can re-use all the mode tracking, registers, marks, test support, etc.). once we have something more close to usable we can revisit that decision if necessary.

**Comment by Suyashtnt** [comment]
I wonder how feasible it would be to embed the helix core into zed and use that for not just helix emulation but just straight up full helix support. Similar to how neovide does it for nvim

**Comment by ericpko** [comment]
I know this isn't a solution, but here are my settings.json and keymap.json files in case anyone searching for this topic wants a quick fix. It's not exactly like helix of course, but decently close. I added as many keymappings from [here](https://docs.helix-editor.com/master/keymap.html) as possible.

I've been using this for about a week and I like it, for now. I would still love to see an actual helix integration into Zed.
[keymap.json](https://github.com/user-attachments/files/15787832/keymap.json)
[settings.json](https://github.com/user-attachments/files/15787836/settings.json)

**Comment by jabcross** [comment]
Great keymap. There are some inconsistencies but it's a great start!

One thing I noticed right away is that the basic word motions (like `e` and `b`)  don't create selections, so you can't delete until the end of the word with `e d` in normal mode for instance. `%` also doesn't select the whole file, etc.

 What do you say of creating a repository and working on this collaboratively?

**Comment by garthtrickett** [comment]
here are a few more ontop of @ericpko 's 
      "w": "vim::NextWordEnd",
      "ctrl-v": "editor::Paste",
      "shift-u": "editor::Redo", // put key-bindings here if you want them to work in normal & visual mode
      "d": "vim::DeleteRight",
      "x": "editor::SelectLine"

**Comment by noahfraiture** [comment]
Hello,
I created the repository to let everyone give some to be the closest possible https://github.com/noahfraiture/zed-helix.git

**Comment by luccahuguet** [comment]
> Hello, I created the repository to let everyone give some to be the closest possible https://github.com/noahfraiture/zed-helix.git @jabcross @garthtrickett @ericpko Currently I put your recommendation and I'm working on it
> 
> It would be great if it was possible to have the settings in multiple files, because currently it's pretty much impossible to directly use this repo as zed configuration file since we all have different preferences

loved using this, you saved my a$$, ty!

**Comment by jabcross** [comment]
> Hello, I created the repository to let everyone give some to be the closest possible https://github.com/noahfraiture/zed-helix.git @jabcross @garthtrickett @ericpko Currently I put your recommendation and I'm working on it
> 
> It would be great if it was possible to have the settings in multiple files, because currently it's pretty much impossible to directly use this repo as zed configuration file since we all have different preferences

Maybe we should have a generator script that generates a file based on a couple questions, kind of like Powerlevel 10k

**Comment by adriangalilea** [comment]
> > Hello, I created the repository to let everyone give some to be the closest possible https://github.com/noahfraiture/zed-helix.git @jabcross @garthtrickett @ericpko Currently I put your recommendation and I'm working on it
> > It would be great if it was possible to have the settings in multiple files, because currently it's pretty much impossible to directly use this repo as zed configuration file since we all have different preferences
> 
> Maybe we should have a generator script that generates a file based on a couple questions, kind of like Powerlevel 10k

As an alternative proposal, if one could point to the helix config file and translated it to zed, that way every-time you make a change in one both change. A piedra-rosetta of sorts, may be overkill and overcomplicated, but I think it would be the comfiest.

**Comment by smltr** [comment]
This might be a bit out there, but has the idea of using modal editors within a terminal tab been discussed? You can currently open a terminal tab right next to a file tab, type hx <filename> and use helix within Zed. If there were a way to just default to opening files in a terminal with a command, that might be an interesting way to easily get helix editing within the greater Zed environment. I have only played around with this, but if it were a built in feature where I could still easily add my files to LLM context I think it might just do it for me. Also could be a hotkey to switch between views? Just thinking out loud here, want to make sure this idea is at least mentioned.

**Comment by jabcross** [comment]
> This might be a bit out there, but has the idea of using modal editors within a terminal tab been discussed? You can currently open a terminal tab right next to a file tab, type hx 'filename' and use helix within Zed. If there were a way to just default to opening files in a terminal with a command, that might be an interesting way to easily get helix editing within the greater Zed environment. I have only played around with this, but if it were a built in feature where I could still easily add my files to LLM context I think it might just do it for me. Also could be a hotkey to switch between views? Just thinking out loud here, want to make sure this idea is at least mentioned.

Quite frankly that seems a little pointless to me. The whole point of this would be being able to use zed's future extension ecosystem, including things that the terminal can't support like multiple concurrent font sizes (for type hints and such), smooth scrolling etc. If I want to use helix directly, what would be the benefit of using it in an integrated terminal instead of in an external one?

**Comment by smltr** [comment]
> > This might be a bit out there, but has the idea of using modal editors within a terminal tab been discussed? You can currently open a terminal tab right next to a file tab, type hx 'filename' and use helix within Zed. If there were a way to just default to opening files in a terminal with a command, that might be an interesting way to easily get helix editing within the greater Zed environment. I have only played around with this, but if it were a built in feature where I could still easily add my files to LLM context I think it might just do it for me. Also could be a hotkey to switch between views? Just thinking out loud here, want to make sure this idea is at least mentioned.
> 
> Quite frankly that seems a little pointless to me. The whole point of this would be being able to use zed's future extension ecosystem, including things that the terminal can't support like multiple concurrent font sizes (for type hints and such), smooth scrolling etc. If I want to use helix directly, what would be the benefit of using it in an integrated terminal instead of in an external one?

I believe you’re right, it probably doesn’t make sense to consider this. I was thinking as someone who has been trying to pick up helix, it would be nice to have a file browser and integrated LLM assistant on top of the helix editor. But there would be too much else missing from this experience.

**Comment by jabcross** [comment]
> LLM

Just so you know, there is a fork of Helix that implements support for Copilot, while the plugin support isn't ready. https://github.com/AlexanderDickie/helix/tree/copilot

And for the file browser, you can always use tmux and one of the ranger variants. But the existing file picker is pretty good too.

**Comment by hamzahmalik101** [comment]
Does anyone know of a way to rebind movement in zed? Here's what I have in helix:

```
[keys.normal]
# Replacing hjkl keybinds
n = "move_char_left"
o = "move_char_right"
i = "move_line_up"
e = "move_line_down"
```

**Comment by ConradIrwin** [comment]
@hamzahmalik101 https://zed.dev/docs/key-bindings and https://zed.dev/docs/vim should get you started

**Comment by nabby27** [comment]
> Zed is pretty awesome and support for the helix bindings would be nice. Can we do a bounty for someone from the community/core to get this running? I am not so skilled myself 😀.
> 
> I can do $100 for anything helix bindings related.

I feel the same way, I think Zed is fantastic and I would like to see this issue resolved! I just created a [$100 bounty](https://app.opire.dev/issues/01J6HJZCK02FVD4XG900FY36EE) on Opire, if anyone wants to join and add more money to the bounty that would be great. I love the project and this way I can contribute my two cents.

**Comment by alphabitserial** [comment]
> Does anyone know of a way to rebind movement in zed? Here's what I have in helix:
> 
> ```
> [keys.normal]
> # Replacing hjkl keybinds
> n = "move_char_left"
> o = "move_char_right"
> i = "move_line_up"
> e = "move_line_down"
> ```

Wow, another Workman layout user! Small world!

**Comment by thiagomajesk** [comment]
I don't know what the current state of this issue is but I'm particularly interested in having a mechanism where you can configure your own set of keybindings for modal editing (be it Vim, Helix, Kakoune, or something else). 

Maybe this belongs to an extension, but having native support would be a killer feature. Don't get me wrong, I think Helix modal editing is great, but I wish that some of the default keybindings were more consistent (there's a huge thread about keymaps on the Helix repository discussing some of the decisions and they don't seem inclined to change any of it).

If we get that and a TUI equivalent interface like in: https://github.com/zed-industries/zed/issues/7493 it would be perfect 😚👌

**Comment by oliverpool** [comment]
> a mechanism where you can configure your own set of keybindings for modal editing

@thiagomajesk did you have a look at @noahfraiture keymap.json? https://github.com/noahfraiture/zed-helix

It should help get you started.

**Comment by thiagomajesk** [comment]
@oliverpool, does this actually work? How can I test it? (It looks like I missed some Zed updates 😅). From what I'm reading this is built on top of the native Vim support which might be a little limiting, but it's a good start. 

I think I would still love to see Zed support a "generic modal editing" and build both Vim and Helix keybindings on top of native Zed commands. Do you know if this is something maintainers are looking into?

**Comment by noahfraiture** [comment]
@thiagomajesk It's a simple configuration file that you need to add to your setup. However, to be honest, it isn't very effective due to the differences between Vim and Helix motion.

Several people want Helix motion to be implemented, and only @maan2003 has done significant work to make it happen. There has been a new PR [#17575](https://github.com/zed-industries/zed/pull/17575) because the previous one was too extensive for a single person to handle, and no one had the courage to continue it.

This new PR should be merged soon, and then everyone will be free to make improvements, as it doesn’t aim to be a complete integration of Helix into Zed.

**Comment by TeriyakiBomb** [comment]
I’ve fallen for helix bindings, but there are still things missing from helix and I prefer gui apps. This would probably be enough to prompt me to move to Zed.

**Comment by hajarrashidi** [comment]
I have paused my use of Zed because I need to improve my Helix skills. I hope there will soon be keybindings for Helix in Zed

**Comment by haras-unicorn** [comment]
thank you for the initial support for helix! :)

i read a bit through this PR and it adds the basic helix motions through a somewhat awkward keymap to enter helix normal mode

not sure if this should be closed yet though because coming from helix people will want way more than just that

**Comment by luccahuguet** [comment]
> thank you for the initial support for helix! :)
> 
> i read a bit through this PR and it adds the basic helix motions through a somewhat awkward keymap to enter helix normal mode
> 
> not sure if this should be closed yet though because coming from helix people will want way more than just that

I believe this calls for a follow-up issue, where we can follow future improvements

is there one already?

**Comment by leroycep** [comment]
Having looked at the vim mode implementation a bit, I'm wondering if it would make more sense to just make a completely separate helix mode. While helix and vim are both modal editors, vim is more distant from non-modal editors like zed because of it's verb-object structure.

In helix, much like in zed, the multi-cursor model is fundamentally how you interact with the buffer. You make list of selections, then perform a command like `delete` or `indent` on that list of selections.

Some other differences I've noticed:

- vim clamps the cursor to a line; helix will wrap onto the next line (like zed does)
- vim has two separate delete commands, 'd' and 'x'. 'd' requires an object, and 'x' just deletes a character. Helix only has 'd', which is identical to pressing the delete key in zed, except that it copies the deleted text into the register `"`
- As far as I know, helix doesn't have an operator stack? Correct me if I'm wrong.
- I don't think helix needs to track the previous mode it was in, or worry about temporary modes in general? Most of the other modes drop back into normal mode, and "space" mode and related act more like a command palette than anything.

Now of course there are some similarities between vim and helix:

- You can use `.` to "replay" some inserted text
- Helix and vim both have the concept of registers
- Pressing `:` in normal mode will let you enter a command
- Pressing `/` in normal mode will enter search mode

But honestly I think a helix's keymap is closer to zed's than it is to vim's.

**Comment by redactedontop** [comment]
Hallo!

Sorry to bother, but any updates?
I'm in love with helix-style keybinds, and I wanna make my whole editor (and life!) use them. A separate helix mode would be definitely prefered.

Thanks,
Alex <3

**Comment by redactedontop** [comment]
Actually, I believe that Helix could be in the same mode as vim, but kakoune bindings would benefit for a different mode (as it doesn't have a visual mode).
Or maybe helix would be in the kakoune mode.

**Comment by jabcross** [comment]
Wouldn't the ideal thing be figuring out a way to make the keybinding and macro system flexible enough to replicate any of the editors through configuration alone? Most modal editor users customize their keybindings anyways.

**Comment by fingersmurphy** [comment]
It isn't trivial though. You're talking about a fundamental rewrite of
basic interaction with the editor. And it would appeal to a fraction of a
fraction of the user base. Don't get me wrong; I'd love it as an option.
But I'm subscribed to see if someone takes it on as a personal project
rather than listen to people say 'oh, you only need to do this...' I've
looked at what it would take and honestly, no way, not for me.

**Comment by redactedontop** [comment]
Fuck it. I'm doing it.

**Comment by fingersmurphy** [comment]
Taking it on as a personal project, surely

**Comment by Diogenesoftoronto** [comment]
I want to know what would be required to get this working, I love the helix editor's keymap and I love using it. I want this. If you want to work with me on this I would definitely be open to that.

**Comment by forivall** [comment]
As mentioned in https://github.com/zed-industries/zed/pull/19175#issuecomment-2543005193 , I have a config that implements enough helix key bindings for my uses for light editing, so that can be useful for people looking for something now

**Comment by infogulch** [comment]
@forivall that is helpful! Does this file include personal/non-helix-related keymap customizations as well?

Watching this thread, I'm noticing that there are many people that are moderately interested in this feature, but it still requires a lot of time to drive it to completion. To channel this energy, I propose we need the following:

1. Zed to support multiple keymap files, so users can update the helix-specific keymap easily without overwriting or manually merging it with their own customizations. See: #16294
1. Create a repo to keep the best current keymap and track unimplemented/unsupported features

I just created https://github.com/infogulch/zed-helix-keymap to hopefully address step 2.

**Comment by forivall** [comment]
@infogulch I think you'll just need to take it as-is, and, if you want correctness, systematically verify that all of the bindings are correct. It was a very ad hoc implementation, so, while I don't try to deviate from vanilla helix, I probably have a few things wrong.

**Comment by ConradIrwin** [comment]
@forivall would you (or anyone) like to tidy up those bindings and add more to the nascent Helix mode in Zed?

Happy to pair on this if you'd like https://cal.com/conradirwin/pairing

**Comment by overby-me** [comment]
@franciskafyi Why is this considered complete? Isn't it still only a WIP implementation?

**Comment by haras-unicorn** [comment]
maybe we should make another issue detailing whats missing since this seems like a basic keymap was made?

**Comment by 42willow** [comment]
- select mode is missing ([helix docs](https://docs.helix-editor.com/keymap.html#select--extend-mode))
- there are [some issues](https://github.com/zed-industries/zed/issues?q=is%3Aissue%20state%3Aopen%20helix%20type%3ABug%20created%3A%3E2025-05-01) with the normal mode keymap
- it would also be nice to include the default helix bindings for insert mode for completeness ([helix docs](https://docs.helix-editor.com/keymap.html#insert-mode))

.. probably not a complete list given that I haven't spent too much time comparing the key bindings and haven't used helix for long

**Comment by roylaurie** [comment]
> maybe we should make another issue detailing whats missing since this seems like a basic keymap was made?

Yeah, maybe an "epic" that acts as a rolling index. Break off smaller units of work into separate issues. This could be the "implement initial support" issue.

**Comment by kasper9n** [comment]
Match mode is missing as well

**Comment by ConradIrwin** [comment]
With #31843, we added an explicit `helix_mode` setting; as noted here though it's *very* incomplete and early.

Can I ask someone who's more familiar with Helix than me to take a pass at a page like https://zed.dev/docs/vim that gives an overview of where our helix mode is at? Maybe @42willow or @haras-unicorn? (docs are in this repo under the "docs" directory).

Unlike vim mode there is no-one at Zed actively working on Helix (though I am still very happy to keep merging in PRs, and helping out), so I want to make it clear to people who aren't active on the issue tracker what to expect. Something like:

```
Zed contains a nascent community-supported Helix implementation. While the basics should work, there are many features missing, and a significant number of smaller differences between how things work in Helix vs Zed. We'd love your help working on these!

## What's working
...

## Recommended settings
...

## What's missing
...

## What needs improvements
...
```

**Comment by haras-unicorn** [comment]
im still switching over to zed from vscode but i wouldnt mind making a list/docs over the weekend when i can mess around with it a bit more

**Comment by ascarter** [comment]
Conrad - would you like a docs page for Helix or a wiki page? I can take a run at state of the world. I use Helix as my terminal editor but I use zed most of the time and prefer the helix bindings. I started to go through the helix tutorial and hit several issues right away (added https://github.com/zed-industries/zed/issues/31997 and https://github.com/zed-industries/zed/issues/32020). Are you looking for individual issues or a summary doc that could work as a punchlist get helix closer. I feel like being able to run the helix tutorial would be a good milestone since It certainly covers the motions and even gets a bit into the extend helix mode support for other operations as well.

**Comment by ConradIrwin** [comment]
Docs page would be great (particularly as it's a real setting now) as I'm mostly worried about making sure new users understand what they're getting themselves in for.

A punch list (a.la. https://zed.dev/channel/vim-393/notes#Vim-Roadmap--Jan-2025) would also be useful for anyone looking for ways to contribute.

**Comment by roylaurie** [comment]
> the helix tutorial would be a good milestone

I love that idea.

The tutorial is just a text file, so a sort of "are we helix yet?" integration test could work.

**Comment by TeriyakiBomb** [comment]
How does one even test helix mode? The config in  #31843 is unclear

**Comment by ConradIrwin** [comment]
We *almost* support all of vim tutor, the only missing item is `U` for "restore current line to before the most recent change" :D. How hard could it be...

@TeriyakiBomb add `{"helix_mode": true}` to your Zed settings (if you're after that commit, v0.193+ - not released yet). If you're before that commit, set `{"vim": {"default_mode": "helix_normal"}}`.

**Comment by TeriyakiBomb** [comment]
Thanks @ConradIrwin 

Ran into an immediate issue with helix tutor `bea` between words inserts after the space of the first word instead of at the cursor - we opening an issue per observation or leaving them inline for now? 

Edit - actually `a` is a bit strange across the board

**Comment by ConradIrwin** [comment]
Best would probably be a GitHub discussion/meta-issue with checkboxes. We know there's a lot to do, so probably not that helpful to have hundreds of GitHub issues.

Not sure if GitHub has anyway of making a checklist that many non-maintainers can edit

**Comment by Malix-Labs** [comment]
### **Zed just released a Helix mode in [0.193.0](https://github.com/zed-industries/zed/releases/tag/v0.193.0-pre)!**

https://zed.dev/releases/preview/0.193.0

https://x.com/Malix_Labs/status/1938874567721881643?t=_rg1nJqHiAGS8v-8WnW9kg&s=19

**Comment by adriangalilea** [comment]
Any plan on adding:
`mi`/`ma`
`gw`

Also, are keybindings customizable?

**Comment by luccahuguet** [comment]
> Best would probably be a GitHub discussion/meta-issue with checkboxes. We know there's a lot to do, so probably not that helpful to have hundreds of GitHub issues.
> 
> Not sure if GitHub has anyway of making a checklist that many non-maintainers can edit

I started it. Let's do this! 


### Are we Helix yet? The definitive meta-discussion to make Helix mode on Zed truly great:
https://github.com/zed-industries/zed/discussions/33580

**Comment by guttermonk** [comment]
Got helix_mode working in zed. Very exciting! Now anyway to change the helix keybinds for a colemak-dh layout (outlined here https://github.com/helix-editor/helix/discussions/5620)?

It would be really slick if helix_mode could automatically pull in the custom keybinds from my helix config.

**Comment by pulsexlb** [comment]
Good news! But seems something wrong in helix mode... some key binding not work in visual mode such as `g e` and `g l`. is it a bug?

**Comment by infogulch** [comment]
This is great! Seeing as the Zed team is accepting patches for helix mode directly, I wonder if https://github.com/infogulch/zed-helix-keymap is useful anymore. Unless someone produces a good argument why the repo could still be useful, I will soon add a notice to the readme pointing to [Are we Helix yet? #33580](https://github.com/zed-industries/zed/discussions/33580) and archive it.

Edit: zed-helix-keymap is now archived.

**Comment by MattCornicDev** [comment]
/attempt

Hi ! I'm going to work on adding a Helix keymap to Zed.

Implementation plan:
- Identify essential Helix commands (e.g. `v`, `d`, `y`, `p`, `u`, `Ctrl-o`, etc.)
- Create a `helix.rs` file with specific bindings
- Integrate the keymap into the existing system (`keymap.rs`)
- Test shortcuts across different modes (normal, visual, insert)

Open to feedback or suggestions before I begin.

**Comment by Malix-Labs** [comment]
@MattCornicDev be sure to check #33580 and all the work that has already been done

**Comment by MattCornicDev** [comment]
@Malix-Labs Thanks a lot for the heads-up!

**Comment by jgarte** [comment]
yaaaaaas! Thanks for adding Helix support. I will start using Zed once this is merged.

**Comment by paaloeye** [comment]
Is [Kakoune](https://github.com/mawww/kakoune) keymap coming next?

**Comment by Malix-Labs** [comment]
@paaloeye you're not the first to ask for it (see #6447 and #33580), feel free to open a different issue about it

**Comment by TeriyakiBomb** [comment]
> Is [Kakoune](https://github.com/mawww/kakoune) keymap coming next?

I mean it's basically the same Keymap just with shift motions on alt and the removal of select mode right?

I've recently moved to Kakoune from Helix and utterly loving it so far

**Comment by janwirth** [comment]
Helix is defo a step back, I've been kakoune since 2018 and I dearly miss it.

**Comment by Malix-Labs** [comment]
This issue is about Helix keymaps.
Please open / upvote a new [feature request](https://github.com/zed-industries/zed/discussions/new?category=feature-requests) about Kakoune to discuss about it.

**Comment by ConradIrwin** [comment]
For people who want Kakoune, to what extent do we need different behavior vs different bindings? If it's mostly another keymap that can re-use vim and helix support, that seems doable; I'm less excited about implementing a bunch of kakoune specific features in rust though

**Comment by luccahuguet** [comment]
> For people who want Kakoune, to what extent do we need different behavior vs different bindings? If it's mostly another keymap that can re-use vim and helix support, that seems doable; I'm less excited about implementing a bunch of kakoune specific features in rust though

it may be helpful to open a discussion for that

it worked out well for helix

**Comment by TeriyakiBomb** [comment]
For the most part, helix and kakoune bindings are almost identical. The biggest difference is that kakoune does not have visual/select mode. You extend selections with the shift key and WEB etc motions are moved to alt.

Other than that, there’s minor behaviour and feature differences that could be addressed, or left out. The above might be a good enough solution for most.

Some of those differences are thing like:
 - match menu invocations are on alt-a/i
 - kakoune `x` by default does not continue to select lines
 - gj/k go to the top/bottom of the file instead of linewise

There’s more but you get the gist. It’s mostly small differences

**Comment by Malix-Labs** [comment]
> This issue is about Helix keymaps. Please open / upvote a new [feature request](https://github.com/zed-industries/zed/discussions/new?category=feature-requests) about Kakoune to discuss about it.

**Comment by atahrijouti** [comment]
@Malix-Labs 
> > This issue is about Helix keymaps. Please open / upvote a new [feature request](https://github.com/zed-industries/zed/discussions/new?category=feature-requests) about Kakoune to discuss about it.

Conrad is in the Zed team, sometimes it's okay not to follow the rules, especially on a closed out issues, and in answer to genuine questions and answers which will make kakoune users happy. Just like we were made happy by receiving helix rust and keymaps real estate in Zed

**Comment by ConradIrwin** [comment]
Sounds good. Happy if someone wants to start a more specific discussion and @-me, or even a PR with a keymap file.

Thanks for your patience Helix people - we'll let your inboxes calm down now.

**Comment by Malix-Labs** [comment]
@RanuK12 on what.

**Comment by jmevel** [comment]
Seriously, those "AI Engineers" are parasites of the open source world
